SingleLinkedList.java

```diff
@@ -0,0 +1,144 @@
+////////////////////////////////////////////////////////////////////////////////
+// Author: Ben Kuschner
+////////////////////////////////////////////////////////////////////////////////
+
+public class SingleLinkedList<T> {
+   
+   /////////////////////////////////////////////////////////
+   // Private link class
+   /////////////////////////////////////////////////////////
+   private class Link<T> {
+      // Member Variables
+      private T data;
+      private Link<T> next;
+
+      // Constructor(s)
+      public Link (T data) {
+         this.data = data;
+         next = null;
+      }
+   
+      // Member methods
+      public T getData() {
+         return this.data;
+      }
+      
+      public boolean hasNext() {
+    	  return next != null;
+      }
+   
+      public Link<T> getNext() {
+         return next;
+      }
+
+      public void setNext(Link<T> next) {
+         this.next = next;
+      }
+
+   } // end of class link
+
+   /////////////////////////////////////////////////////////
+   // Single Linked List
+   /////////////////////////////////////////////////////////
+
+   // Member variables
+   private Link<T> first;
+	
+   // Constructor(s)
+   public SingleLinkedList() {
+      first = null;
+   }
+
+   // Member methods
+	
+   public boolean isEmpty () {
+      return first == null;
+   }
+	
+   // Inserts a node at the front of the list.
+   public void insertFront(T data) {
+      Link<T> newLink = new Link<T>(data);
+
+      newLink.setNext(first);
+      first = newLink;
+   }
+
+   // Inserts a node at the back of the list
+   public void insertBack(T data) {
+      Link<T> last = first;
+      while (last.hasNext()) { //sets last to the last node
+    	  last = last.getNext();
+      }
+      last.setNext(new Link<T>(data));
+   }
+
+   // Returns the amount of objects in the list.
+   public int length() {
```

Comment:
I would consider storing a length member variable so that you do not need to linearly scan the list each time you call length.

For example:

for (int index = 0; index < list.length(); ++index) { ... } could take significantly large amounts of time because this scans the entire list (especially if the list is large).